File: 844-backspace-string-compare/844-backspace-string-compare.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        
        def convert(s):
            stack = []
            for c in s:
                if c != '#':
                    stack.append(c)
                else:
                    if stack:
                        stack.pop()
            return ''.join(stack)
        
        return convert(s) == convert(t)
        
        

File: 844-backspace-string-compare/844-backspace-string-compare.cpp
class Solution {
public:
    
    string convert(string s)
    {
        stack <char> st;
        string compare = "";
        for(char c : s)
        {
            if(c != '#')
                st.push(c);
            else
                if(!st.empty())
                    st.pop();
        }
        while(!st.empty())
        {
            compare += st.top();
            st.pop();
        }
        return compare;
    }
    
    bool backspaceCompare(string s, string t) 
    {
        return convert(s) == convert(t);
    }
};